### PR TITLE
Cherry pick of #100690: azure: fix node public IP not able to fetch issues from IMDS

### PR DESCRIPTION
**Pull request by nilo19**
Cherry pick of #100690 on release-1.21.

#100690: azure: fix node public IP not able to fetch issues from IMDS

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

/triage accepted
/priority critical-urgent
/sig cloud-provider
/area provider/azure
/kind bug

cc @feiskyer

### CODE REVIEW

**Review by feiskyer — APPROVED** [review]
/lgtm

**Comment by feiskyer** [comment]
@andrewsykim  @cheftako could you help to approve this cherry-pick?

**Comment by cheftako** [comment]
Happy to review/approve once #100690 merges.

**Comment by feiskyer** [comment]
@cheftako could you help to approve 100690?

/retest

**Comment by cheftako** [comment]
/lgtm
Happy to approve once #100690 merges (I already approved it). Would suggest you line up you release manager for the cherry pick as I believe you also need their approval.

**Pull request merged**
